lib/internal/process/per_thread.js

```diff
@@ -148,6 +149,46 @@ function wrapProcessMethods(binding) {
     };
   }
 
+  const threadCpuValues = new Float64Array(2);
+
+  // Replace the native function with the JS version that calls the native
+  // function.
```

Comment:
Why don't we just implement the whole thing in C++? I’m a bit confused about this particular comment.

Reply:
Well, two reasons:

1. I'm not totally familiar with C++ :)
2. I want to keep the most parts we can in JS so that will be easier for a bigger chunk of contributors to chime in.